odoo/tools/set_expression.py

```diff
@@ -137,6 +137,31 @@ def __get_leaf(self, ref: str | int, raise_if_not_found: bool = True) -> Leaf:
             return Leaf(UnknownId(ref), ref)
         return self.__leaves[ref]
 
+    def get_superset_ids(self, ids: Iterable[int]) -> set[int]:
+        return {
+            sub_id
+            for id_ in ids
+            if id_ in self.__leaves
+            for sub_id in self.__leaves[id_].supersets - {id_}
+            if sub_id != id_
```

Comment:
Same for other functions:

No need to exclude `id_` twice and docstring missing.

```suggestion
            for sub_id in self.__leaves[id_].supersets
            if sub_id != id_
```